discojs/src/convertors.ts

```diff
@@ -0,0 +1,126 @@
+import { PreTrainedTokenizer } from "@xenova/transformers";
+import { List, Repeat, Seq } from "immutable";
+import { Image } from "./dataset/image.js";
+
+export function convertToNumber(raw: string): number {
+  const num = Number.parseFloat(raw);
+  if (Number.isNaN(num)) throw new Error(`unable to parse "${raw}" as number`);
+  return num;
+}
+
+export function extractColumn(
+  row: Partial<Record<string, string>>,
+  column: string,
+): string {
+  const raw = row[column];
+  if (raw === undefined) throw new Error(`${column} not found in row`);
+  return raw;
+}
+
+export function indexInList(element: string, elements: List<string>): number {
+  const ret = elements.indexOf(element);
+  if (ret === -1) throw new Error(`${element} not found in list`);
+  return ret;
+}
+
+function isArrayOfNumber(raw: unknown): raw is number[] {
+  return Array.isArray(raw) && raw.every((e) => typeof e === "number");
+}
+
+/** Tokenize and truncates input strings
+ *
+ * @param length number of tokens
+ * @returns encoded string in an array of token, size of max_length
+ **/
+export function tokenizeAndLeftPad(
```

Comment:
Right now this function is only used by the cli and not by the trainingInformation preprocessing, is this intened?

Reply:
for now yes, I didn't wanted to modify too much the internals of discojs. a rewrite of preprocessing is baking in the next PR.